services/cloudapi/cloudapiserver/server.go

```diff
@@ -121,6 +179,80 @@ func (ss *Service) UpdateVmState(ctx context.Context, datacenterID, serverID, ne
 	return nil
 }
 
+func (ss *Service) UpdateBootDevice(ctx context.Context, datacenterID, serverID, newBootDeviceID string) error {
+	oldBootDeviceID, oldBdType, err := ss.GetCurrentBootDeviceID(ctx, datacenterID, serverID)
+	if err != nil {
+		return err
+	}
+	if strings.EqualFold(oldBootDeviceID, newBootDeviceID) {
+		return nil
+	}
+
+	newBdType := BootDeviceTypeCDROM
+	_, apiResponse, err := ss.Client.ImagesApi.ImagesFindById(ctx, newBootDeviceID).Execute()
+	if err != nil {
+		if !(apiResponse != nil && apiResponse.StatusCode == 404) {
```

Comment:
```suggestion
		if !(apiResponse != nil && apiResponse.StatusCode == 404) {
```
		if !apiResponse.HttpNotFound() {

Reply:
solved in - [90b61bb](https://github.com/ionos-cloud/terraform-provider-ionoscloud/pull/507/commits/90b61bb22669e6c0866d253d4535805c673b7c73)